index.js

```diff
@@ -458,15 +459,42 @@ class ClientServiceBase {
   /**
    * Execute a registered contract
    * @param {string} contractId
-   * @param {Object} argument
-   * @param {Object} [functionArgument=undefined]
+   * @param {Object|string} contractArgument
+   * @param {Object|string} [functionArgument=null]
+   * @param {string} [functionId=""]
+   * @param {string} [nonce=null]
    * @return {Promise<ContractExecutionResult|void|*>}
    * @throws {ClientError|Error}
    */
-  async executeContract(contractId, argument, functionArgument) {
+  async executeContract(
+      contractId,
+      contractArgument,
+      functionArgument = null,
+      functionId = '',
+      nonce = null,
+  ) {
+    if (functionArgument === null) {
+      functionArgument = typeof contractArgument === 'object' ? {} : '';
+    }
```

Comment:
This is defensive programming.
If the user doesn't pass the function argument, an empty value is given according to the type of the contract argument.